src/graph/graph.c

```diff
@@ -1383,6 +1400,17 @@ GrB_Matrix Graph_GetRelationMatrix(const Graph *g, int relation_idx) {
 	}
 }
 
+// Returns wether the matrix contains multi edge.
+bool Graph_RelationshipContainsMultiEdge(const Graph *g, int r) {
+	ASSERT(array_len(g->stats.edge_count) > r
```

Comment:
`array_len(g->stats.edge_count)` this is tinkering with the internals, either remove this check or introduce a function to statistics which will let you know how many relationship types it tracks